core/src/main/kotlin/transformers/descriptors/DefaultDescriptorToDocumentationTranslator.kt

```diff
@@ -70,6 +72,19 @@ class DokkaDescriptorVisitor(
             listOfNotNull(descriptorData),
             mutableSetOf() // TODO Implement following method to return proper results getXMLDRIs(descriptor, descriptorData).toMutableSet()
         )
+
+        return if (descriptor.kind == org.jetbrains.kotlin.descriptors.ClassKind.ENUM_CLASS)
+            visitEnumDescriptor(descriptor, parent, clazz)
+        else clazz
+    }
+
+    fun visitEnumDescriptor(descriptor: ClassDescriptor, parent: DRI, clazz: Class): ClassLike {
+        val dri = parent.withClass(descriptor.name.asString())
+        val scope = descriptor.getMemberScope(emptyList())
+        val enum_values = scope.classes(dri).filter { it.kind == KotlinClassKindTypes.ENUM_ENTRY }
+            .map { EnumEntry(it) }
+
+        return Enum(clazz, enum_values)
     }
```

Comment:
If you create a separate method for visiting an EnumDescriptor (which is a `ClassDescriptor` BTW), then the original `visitClassDescriptor` should:
1) be renamed to `visitClassLikeDescriptor` (don't like this naming though, as `ClassLikeDescriptor` does not exist)
2) only dispatch the logic to two separate methods

or you shouldn't create this separate method at all

Reply:
Unfortunately this method is override of function from library, so I can't rename it.